File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const map = new Map();
        let longest = 0;
        for(let i = 0; i<nums.length; i++){
            let val = nums[i];
            let targetLongest = 1;
            if(map.has(val-1)){
                targetLongest = map.get(val-1)+1;
                map.set(val,targetLongest);
            } else{
                map.set(val, 1);
                if(longest==0) {
                    longest = 1;
                }
            }
            while(map.has(++val)){
                targetLongest++;
                map.set(val,targetLongest);
            }
            if(targetLongest>longest){
                longest = targetLongest;
            }
        }
        
        return longest;
    }
}

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l=0
        r=len(nums) - 1
        
        while l<=r:
            m = (l+r)//2
            val = nums[m]
            if val == target:
                return m
            if val < target:
                l = m + 1
            if target < val:
                r = m - 1

        return -1    